### PR TITLE
Added BreadthFirstSearch for Java

### PR DESCRIPTION
**Pull request by PranayChauhan2516**
Added BreadthFirstSearch for Java.

### CODE REVIEW

**Review by SobhanBera — COMMENTED** [review]
Please add the credits to geeks for geeks into the top of code.

File: algorithms/java/BreadthFirstSearch.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    void BFS(int s) { 
+        boolean visited[] = new boolean[V];        			 
+        LinkedList<Integer> queue = new LinkedList<Integer>(); 
+        visited[s]=true; 
+        queue.add(s); 
+        while (queue.size() != 0) { 
+            s = queue.poll(); 
+            System.out.print(s+" ");
+            Iterator<Integer> i = adj[s].listIterator(); 
+            while (i.hasNext()) {
```
Comment (SobhanBera) [inline algorithms/java/BreadthFirstSearch.java:32]: If you are copying from another website then please provide the link to that website.

Comment (PranayChauhan2516) [reply algorithms/java/BreadthFirstSearch.java:32]: This is mainly done by me, I just took an overview from gfg. Though I'll make the changes if you insist.

**Comment by PranayChauhan2516** [comment]
Done @SobhanBera

File: algorithms/java/BreadthFirstSearch.java (hunk 1 above)
Comment (SobhanBera) [reply algorithms/java/BreadthFirstSearch.java]: Great Thanks for your contribution.

**Comment by SobhanBera** [comment]
Thanks for your contribution.

**Pull request merged**